essentials-module-teleport/src/main/kotlin/io/github/nyliummc/essentials/commands/RTPCommand.kt

```diff
@@ -59,7 +60,8 @@ object RTPCommand {
         var newY = 0.0
         var i = world.seaLevel
 
-        while (!world.isAir(BlockPos(newX, i++.toDouble(), newZ)) && newY + world.seaLevel < world.height) {
+        val pos = BlockPos(newX, i++.toDouble(), newZ)
+        while ((!world.isAir(pos) || world.getBlockState(pos).block == Blocks.CAVE_AIR) && newY + world.seaLevel < world.height) {
```

Comment:
Infinite loop since pos never updates, and cave air doesn't exist except during chunk gen.